### PR TITLE
Removing maintainer, adding user_hdx to api key config

### PR DESCRIPTION
**Pull request by willdoran**
This pull request makes the following changes:
-

Testing checklist:
- [ ]

- [ ] I certify that I ran my checklist

Fixes ushahidi/platform#2986

Ping @ushahidi/platform

### CODE REVIEW

**Comment by rjmackay** [comment]
@willdoran I know there's lots of detail in the spec on #2986 but can we make sure pull requests still actually have a relevant description? and if not completing section - ie. test checklist - either delete them or reference a checklist elsewhere?

File: app/settings/user-settings/user-settings.html [hunk 1]
```diff
@@ -40,39 +40,66 @@ <h2 class="form-sheet-title" translate="settings.settings_list.user_settings">
             </div>
 
             <img class="field-logo" src="/img/hdx-logo.png" />
+            <form name="api">
+                <div
+                    class="form-field init required"
+                    ng-class="{'error': api.$invalid && api.$dirty,'success': !api.$invalid && api.$dirty}"
+                    ng-show="!hxlUserId"
```
Comment (rjmackay) [inline app/settings/user-settings/user-settings.html]: Can we rename `hxlUserId` to something meaningful? I spent a few minutes wonder what was going on till I realised its really a show/hide toggle, not the actually userId Model field

File: app/settings/user-settings/user-settings.html [hunk 2]
```diff
@@ -40,39 +40,66 @@ <h2 class="form-sheet-title" translate="settings.settings_list.user_settings">
...
-            <div
-                class="form-field init required"
-                ng-class="{'error': api.$invalid && api.$dirty,'success': !api.$invalid && api.$dirty}"
-                ng-show="!hxlApiKey"
-            >
-                <form name="api">
+                <div
+                    class="form-field init required"
+                    ng-class="{'error': api.$invalid && api.$dirty,'success': !api.$invalid && api.$dirty}"
+                    ng-show="!hxlApiKey"
```
Comment (rjmackay) [inline app/settings/user-settings/user-settings.html]: Same with this field - it really needs a meaningful name

File: app/settings/user-settings/user-settings.html (hunk 1 above)
Comment (rjmackay) [reply app/settings/user-settings/user-settings.html]: Should the ng-show check include invalid and dirty checks here? (the same as for the api key below)

File: app/settings/user-settings/user-settings.html (hunk 2 above)
Comment (willdoran) [reply app/settings/user-settings/user-settings.html]: Yeah, there were the names it already had but I'll change them to explain they're bools.

**Comment by willdoran** [comment]
@rjmackay Yep generally I will add the description, this was just ready for code review on Friday not for testing yet.

File: app/settings/user-settings/user-settings.html (hunk 1 above)
Comment (willdoran) [reply app/settings/user-settings/user-settings.html]: Yes I think that's a good idea, I'll add it.

File: app/settings/user-settings/user-settings.controller.js [hunk 3]
```diff
@@ -34,29 +52,52 @@ function (
...
 
     function saveKey() {
-        $scope.api.$dirty = true;
-        if ($scope.apiKey) {
-            UserSettingsEndpoint.saveCache({id: $scope.hxlApiId,  user_id: $rootScope.currentUser.userId,  config_key: 'hdx_api_key', config_value: $scope.apiKey}).$promise.then((response) => {
-                $scope.hxlApiKey = '*** *** *** *** *** *** *** ' + $scope.apiKey.slice($scope.apiKey.length - 4);
-                $scope.hxlApiId = response.id;
-                Notify.notifyAction('settings.user_settings.api_key_saved', null, false, 'thumb-up', 'circle-icon confirmation', {callback: goToHdxView, text: 'settings.user_settings.start_tagging', callbackArg: null, actionClass: 'button button-alpha'});
-            });
+        var calls = [];
```
Comment (rowasc) [inline app/settings/user-settings/user-settings.controller.js]: I think I found a bug. 
Prerequisite
- [ ] having the API key and maintainer id set up
Steps
- [ ]  change your maintainer id, save
    - [ ] you will receive a validation error but it won't show up (only in network/console). Example : 
{"errors":[{"status":422,"message":"Failed to validate user_settings entity:\nThe user id, 2, and config key, hdx_user_id, already exist"},{"status":422,"title":"The user id, 2, and config key, hdx_user_id, already exist","message":"The user id, 2, and config key, hdx_user_id, already exist","source":{"pointer":"\/config_key"}}]}

File: app/common/locales/en.json [hunk 4]
```diff
@@ -959,9 +959,11 @@
             "user_settings_desc_2": "Once set up, this will enable you to upload data directly to a Humanitarian Data Exchange (HDX) account.",
             "user_settings_desc_3": "You can find your API key on your Humanitarian Data Exchange (HDX) profile once you're logged in.",
             "view_guide": "View the guide to your API key",
+            "hdx_maintainer_id" : "HDX Maintainer Id",
             "api_key": "Api key",
             "profile_page": "Found on your Humanitarian Data Exchange (HDX) profile page",
             "valid_key": "A valid key is required",
+            "valid_user": "A valid user id is required",
```
Comment (rowasc) [inline app/common/locales/en.json]: should be "A valid maintainer id is required"

File: app/settings/user-settings/user-settings.controller.js [hunk 5]
```diff
@@ -34,29 +52,52 @@ function (
...
+        if ($scope.api.hdx_maintainer_id.$dirty) {
+            calls.push(
+                UserSettingsEndpoint.saveCache($scope.hdxSettings.hdx_user_id).$promise
+            );
         }
+
+        $q.all(calls).then((response) => {
+            $scope.hxlMaintainerSet = true;
+            $scope.hxlApiKeySet = true;
+            $scope.hxlApiKey = '*** *** *** *** *** *** *** ' + $scope.hdxSettings.hdx_api_key.config_value.slice($scope.hdxSettings.hdx_api_key.config_value.length - 4);
```
Comment (rowasc) [inline app/settings/user-settings/user-settings.controller.js]: this will probably fail the lint validation

Comment (willdoran) [reply app/settings/user-settings/user-settings.controller.js]: which linter can I test this with?

Comment (rowasc) [reply app/settings/user-settings/user-settings.controller.js]: ehh. ignore me. It passes. 🤦‍♀️ IDK but it looked like way more than 100 chars to me : ( sorry

**Pull request merged**